### DISCUSSION TITLE
Run DotNetTest() with sudo

### DISCUSSION THREAD

**Discussion by zplan** [cake-build/cake#3805] [Q&A]
My Cake 2.0 runs on a debian and my tests need sudo permission to pass. I don't want to run the whole cake process with sudo.

How can I execute just the DotNetTest() command with sudo? Is this doable at all?

Thanks in advance for feedback!

**Comment by nick5454** [comment]
You can probably just do this. I use these for quick bash or cmd call

`
    $"sudo whatever call you need".BashOut();

public static string BashOut(this string cmd)
{
	var escapedArgs = cmd.Replace("\"", "\\\"");
	   
	Console.WriteLine("Cmd: " + cmd);

	var process = new Process()
	{
		StartInfo = new ProcessStartInfo
		{
			FileName = "/bin/bash",
			Arguments = $"-c \"{escapedArgs}\"",
			RedirectStandardOutput = true,
			UseShellExecute = false,
			CreateNoWindow = true,
		}
	};
	process.Start();
	string result = process.StandardOutput.ReadToEnd();
	process.WaitForExit();
	return result;
}

public static string CommandLineOut(this string cmd)
{
	var escapedArgs = cmd.Replace("\"", "\\\"");
	   
	Console.WriteLine("Cmd: " + cmd);

	var process = new Process()
	{
		StartInfo = new ProcessStartInfo
		{
			FileName = "cmd",
			Arguments = $"-c \"{escapedArgs}\"",
			RedirectStandardOutput = true,
			UseShellExecute = false,
			CreateNoWindow = true,
		}
	};
	process.Start();
	string result = process.StandardOutput.ReadToEnd();
	process.WaitForExit();
	return result;
}`

**Comment by zplan** [comment]
Thanks @nick5454 
This is how I do it currently as well, but as you can see it looks much more complex and I can't use the Cake "DotNetTest" command at all.
In the end Cake is also "just" starting a process, but it looks like there is no built-in solution for sudo.
